### PR TITLE
Clowns and Mimes acts no longer get ruined when you examine their ID cards

### LINKED ISSUE #40560: ECONOMY: Clown and probably mime IDs display your main character name upon examine, not clown name
" That's Clunwe's ID Card (Clown)
A card used to provide ID and determine access across the station.
It is a tiny item.
The registered account belongs to 'Fortune Ray' and reports a balance of $972.
The Service Budget reports a balance of $12823.3."

@iamgoofball fix your shit

### PR DESCRIPTION
**Pull request by ghost**
:cl: Floyd / Qustinnus
fix: Clown and Mimes their ID cards no longer show the name of some random weirdo.
/:cl:

fixes #40560

also removed some operations that dont need to happen in the first place

### CODE REVIEW

File: code/modules/jobs/job_types/job.dm [hunk 1]
```diff
@@ -81,10 +81,8 @@
 	if(!H)
 		return FALSE
 	if(!visualsOnly)
-		var/datum/bank_account/bank_account = new(H.real_name, src)
-		bank_account.account_holder = H.real_name
-		bank_account.account_job = src
-		bank_account.account_id = rand(111111,999999)
+		var/obj/item/card/id/I = H.wear_id ? H.wear_id.GetID() : null
+		var/datum/bank_account/bank_account = new(I.registered_name, src)
```
Comment (ShizCalev) [inline code/modules/jobs/job_types/job.dm]: don't think this has handling for null values

Comment (ShizCalev) [reply code/modules/jobs/job_types/job.dm]: https://github.com/tgstation/tgstation/blob/0ebfc529f50bd6dd7f90ac311b3e97e787e6c8ca/code/modules/economy/account.dm#L9-L12

**Comment by DaxDupont** [comment]
Golfball man saved by local Twink

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
What Shiz said, doesn't handle possibility of job outfit not including an ID. Not common but better to handle than not.

**Comment by SpaceManiac** [comment]
```
code\modules\jobs\job_types\job.dm:89:error: bank_account.payday: undefined var
code\modules\jobs\job_types\job.dm:90:error: bank_account.account_id: undefined var
code\modules\jobs\job_types\job.dm:86:warning: bank_account: variable defined but not used
code\modules\jobs\job_types\job.dm:88:warning: bank_account: variable defined but not used
```

**Comment by Qustinnus** [comment]
wat howd that happen

**Comment by Qustinnus** [comment]
oh lol im triple dumb

**Pull request merged**